### PR TITLE
Speed up property unmarshalling

### PR DESCRIPTION
**Pull request by dirkkul**
### What's being changed:

Needs https://github.com/weaviate/weaviate/pull/5192

This refactors GET requests to use more efficient unmarshalling already in place for Aggregate. This uses:
- a more efficient library that does not need allocations
- only unmarshalls what is actually requested

There were a few things that needed to be changed:
- Some modules request extra properties, these need to be added to the props that are extracted
- Some modules expect all properties to be present - in this case all props need to be extracted
- There are many many code-paths that lead to the unmarshalling. I changed everything using `FromBinaryOptional` to use the need unmarshalling. There is also `FromBinary` which we can maybe migrate later.

It also adds more tests using the generative and rerank dummy so we can be sure things are not breaking here

~~Test script with sphere dataset: https://gist.github.com/dirkkul/9fce2cfe67f8408d3207d63b6754aab2 that reads 2000 queries with all properties:~~ (measured with an inefficient version of the multi target search)

~~New:  4.772850275039673
Old: 7.952303886413574~~

Microbenchmark

(old code) BenchmarkUnmarshalPropertiesFullObject-10         	      3061 ns/op. 
(new code, all properties) BenchmarkUnmarshalPropertiesExplicitAll-10 	      2049 ns/op
(new code, selected properties) BenchmarkUnmarshalPropertiesExplicitOnlySome-10  	      1128 ns/op

### Review checklist

- [x] Documentation has been updated, if necessary. Link to changed documentation:
- [x] Chaos pipeline run or not necessary. Link to pipeline:
- [x] All new code is covered by tests where it is reasonable.
- [x] Performance tests have been run or not necessary.

### CODE REVIEW

**Review by parkerduckworth — COMMENTED** [review]
Seems to have some failing acceptance tests related to the changes

**Review by amourao — DISMISSED** [review]
LGTM, I'll also do some benchmarking to see how it impacts query performance.

File: adapters/handlers/graphql/local/get/class_builder_fields.go [hunk 1]
```diff
@@ -17,6 +17,8 @@ import (
 	"regexp"
 	"strings"
 
+	additional2 "github.com/weaviate/weaviate/usecases/modulecomponents/additional"
```
[the remark this answers is not in the document]
Comment (dirkkul) [reply adapters/handlers/graphql/local/get/class_builder_fields.go]: done!

Under modulecomponents are things that are shared by different modules. I renamed it to `moduleadditional` to make it more obvious!

**Review by amourao — DISMISSED** [review]

**Review by amourao — DISMISSED** [review]
LGTM

**Review by amourao — APPROVED** [review]

**Pull request merged**